scripts/extensions/auto-tagging-widget/src/auto-tagging.tsx

```diff
@@ -388,8 +389,7 @@ export function getAutoTaggingComponent(superdesk: ISuperdesk, label: string) {
                                 const others = items.filter((tag) => isEntity(tag) === false);
                                 const othersGrouped = others.groupBy((tag) => tag.group.value);
                                 const othersGroupedAndSorted = othersGrouped.sortBy(
-                                    (_, key) => key!.toString().toLocaleLowerCase(),
-                                    (a, b) => a.localeCompare(b),
+                                    (_, key) => this.iMatricsFields[key].order ?? -1,
```

Comment:
I have added sorting in order to prevent groups switching places after an update from the server. For this to work, sorting has to be deterministic and return the same order given the same items.

It looks like your implementation would only work if all possible fields are configured. Can you improve it so it works both without configuration or with partial configuration(for example 2 fields configured out of total of 5)?

Reply:
The back-end returns an object, not an array so it can't be sorted. We could assume that if you iterate object keys, the order would be the same as in JSON that the object was constructed from, but it will not always be the case.

What you can do is to apply sorting by name, that was implemented before in this file, and sort it again using `iMatricsFields` config.

Reply:
yep, that was my idea also. Tnx!